inc/saiport.h

```diff
@@ -572,6 +587,15 @@ typedef enum _sai_port_attr_t
     SAI_PORT_ATTR_INTERNAL_LOOPBACK_MODE,
 
     /**
+     * @brief Forward Error Correction (FEC) control
+     *
+     * @type sai_port_fec_mode_t
+     * @flags CREATE_AND_SET
+     * @default SAI_PORT_FEC_MODE_NONE
```

Comment:
Yes, I was thinking that the list will fit the case where there is a hw ability to perform any of the supported modes based on the peer's mode. We are leaving it to the HA to configure the valid FEC mode for a port speed. So, any mis-configuration would lead the port to remain down. Isnt it? We can have this as the static enum now if the case i mentioned is not really useful.